besu/src/main/java/org/hyperledger/besu/RunnerBuilder.java

```diff
@@ -558,7 +560,12 @@ public Runner build() {
   }
 
   private Optional<NatManager> buildNatManager(final NatMethod natMethod) {
-    switch (natMethod) {
+
+    final NatMethod detectedNatMethod =
+        Optional.of(natMethod)
+            .filter(not(isEqual(NatMethod.NONE)))
```

Comment:
Do you need this filter since you handle the case of `NatMethod.NONE` in the switch below?

Reply:
yes the filter is in my opinion necessary. the goal is to try to detect the NAT environment automatically in the case where no NAT environment is given by the user. So when natMethod is equal to NONE I try to automatically detect the NAT environment.  This search can also return NONE if nothing is found.

- No NAT environment is given by the user (NatMethod == Default Value == NONE)
   - Step 1 Empty the optional because the value is equal to NONE
   - Step 2 Optional is empty so we are trying to detect the Nat Environment automatically 
   - Step 3 We recover the detected Nat environment
   - Step 4 The swith returns the good NatManager (if the value is still equal to NONE it is that we have not detected any nat environment)

- NAT environment is given by the user (ex. NatMethod == UPNP)
   - Step 1 Do not Empty the optional because the value is not equal to NONE
   - Step 2 The swith returns the good NatManager

Reply:
So if I understand correctly the user configuring NONE as the nat method explicitly would change nothing compared to if they didn't?

Reply:
yes indeed it can be strange for the user.  I can add an additional natMethod which will be AUTO.  this natMethod will be the default.  if the user chooses NONE there will be no automatic detection. what do you think of that ?

Reply:
I just pushed my modification. I'll let you tell me what you think. I think that you are right and that it allows not to have a strange behavior with NONE

Reply:
Cool! I like the addition of `AUTO`. If the user says `NONE` explicitly, we shouldn't ignore them and go ahead and detect NAT.